### ISSUE TITLE
Check for spelling mistakes in the comments of source code.

### ISSUE THREAD

**Issue by tomfitzhenry** [leotaku/flycheck-aspell#11]
flyspell has a mode for prog-mode, `flyspell-prog-mode`. It'd be great `flycheck-aspell` came with a flymake-diagnostic-function for prog-mode.

A concrete example.

# Steps to reproduce

With `(flymake-aspell-setup)`, write the following code:

```go
package main

// I can't spell hipopopopoatmus.
func main() {

}
```

# Expected

"hipopopopoatmus" should be the only flymake diagnostic.

# Actual

"hipopopopoatmus" and "func" are displayed as flymake diagnostics.

This is a false positive, because "func" is a valid Go keyword.

# Workaround

`(add-hook 'prog-mode-hook 'flyspell-prog-mode)`

**Comment by leotaku** [comment]
`flymake-aspell` actually already comes with something like that, using Aspell's built-in support for syntax filters. It's just that I rarely use the Go language and thus have not added it to the list of supported languages.

I'll add Go to the list of supported languages in the next commit, I think that should solve your issue.

**Comment by leotaku** [comment]
I also understand that it is a bit inelegant that new languages have to be manually added to a list. Maybe, when I get the time, I could implement `flyspell-prog-mode`-equivalent functionality by using a combination of Aspell's context filter functionality and the `comment-start`/`comment-end` variables.

**Comment by tomfitzhenry** [comment]
> I also understand that it is a bit inelegant that new languages have to be manually added to a list.

Elegance is overrated! :) Thanks for the great mode, and the fix!
